lib/tpm_policy.c

```diff
@@ -0,0 +1,254 @@
+//**********************************************************************;
+// Copyright (c) 2015, Intel Corporation
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice,
+// this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice,
+// this list of conditions and the following disclaimer in the documentation
+// and/or other materials provided with the distribution.
+//
+// 3. Neither the name of Intel Corporation nor the names of its contributors
+// may be used to endorse or promote products derived from this software without
+// specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+//**********************************************************************;
+#include <errno.h>
+#include <limits.h>
+
+#include "../lib/tpm2_util.h"
+#include "files.h"
+#include "log.h"
+#include "tpm_session.h"
+#include "tpm_hash.h"
+#include "tpm_policy.h"
+
+static unsigned get_size_from_alg(TPMI_ALG_HASH hashAlg) {
+    switch (hashAlg) {
+        case TPM_ALG_SHA1:
+            return SHA1_DIGEST_SIZE;
+            break;
+        case TPM_ALG_SHA256:
+            return SHA256_DIGEST_SIZE;
+            break;
+        case TPM_ALG_SHA384:
+            return SHA384_DIGEST_SIZE;
+            break;
+        case TPM_ALG_SHA512:
+            return SHA512_DIGEST_SIZE;
+            break;
+        case TPM_ALG_SM3_256:
+            return SM3_256_DIGEST_SIZE;
+            break;
+        default:
```

Comment:
Is this case possible? Should this case actually cause an abort() because something is so wrong?